### PR TITLE
TECH-149: Create test configuration for "/vouchers/voucherstatuscheck/{voucherserialnumber}" PATCH

### PR DESCRIPTION
**Pull request by mgrochalskisoldevelo**
Created mocks and Cucumber tests for PATCH /vouchers/voucherstatuscheck/{voucherserialnumber} endpoint

## Description
Create test configuration for /vouchers/voucherstatuscheck/{voucherserialnumber} endpoint (PATCH method).

TICKET: https://govstack-global.atlassian.net/browse/TECH-149

### CODE REVIEW

File: test/features/support/voucher_cancelation.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+Then('The user successfully cancels the voucher', async () => {
+  await specVoucherCancelation.toss();
+  specVoucherCancelation.response().should.have.status(200);
+  specVoucherCancelation.response().should.have.jsonLike({
+    'message': 'Voucher cancelled'
+  });
+});
+
+// Scenario: The user is not able to cancel a voucher with voucher serial number 4321
```
Comment (dborowiecki) [inline test/features/support/voucher_cancelation.js:37]: I think we should change this test scenario. Reason of test case failure is not voucher number but missing body. Just want to make sure if that's a convention we want to take. With regard do our recent discussion about configurable payloads it seems that including data like serial numbers in test case names is not direction we should take. 
Can we use something like: 
```suggestion
// Scenario: The user is not able to cancel a voucher without providing voucher details.
```
instead?

This is the case for other test cases in scope of this ticket as well.

File: test/features/support/voucher_cancelation.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+Given('The user wants to cancel a voucher with voucher serial number 4321', () => {
+  serialNumberParam = '4321';
+  return serialNumberParam;
+});
+
+When('The user triggers an action without payload to cancel a voucher', () => {
+  specVoucherCancelation.patch(baseUrl).withBody()
```
Comment (dborowiecki) [inline test/features/support/voucher_cancelation.js:44]: I see that in this test case you don't use requestFunction. 
Can you take same approach with setting up body value that you take with  `voucherserialnumber` query param? Body variable would be defined in `@Given` statement and used by requestFunction accordingly.

**Review by dborowiecki — CHANGES_REQUESTED** [review]
Please look into comment's about using specified voucher numbers in test case scenarios. 
While I think it can be parametrized in @Given statement we cannot do it in the test scenario name. Additionally it's not very descriptive from report perspective (it is not straightforward to comprehend that a given voucher code means success/failure in the test scenario).

File: test/features/support/voucher_cancelation.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+const { Given, When, Then, Before, After } = require('@cucumber/cucumber');
+const { localhost } = require('./helpers/helpers');
+
+let specVoucherCancelation;
+let serialNumberParam;
+
+const baseUrl = `${localhost}vouchers/voucherstatuscheck/{voucherserialnumber}`;
+const requestFunction = () =>
+  specVoucherCancelation.patch(baseUrl).withBody({
+    'voucherserialnumber': `${serialNumberParam}`,
```
Comment (KarolinaKopacz) [inline test/features/support/voucher_cancelation.js:11]:
```suggestion
    'voucherserialnumber': serialNumberParam,
```

File: test/features/support/voucher_cancelation.js (hunk 2 above)
Comment (KarolinaKopacz) [reply test/features/support/voucher_cancelation.js:44]: Good idea!

File: test/features/support/voucher_cancelation.js [hunk 4]
```diff
@@ -0,0 +1,117 @@
+const pactum = require('pactum');
+const { Given, When, Then, Before, After } = require('@cucumber/cucumber');
+const { localhost } = require('./helpers/helpers');
+
+let specVoucherCancelation;
+let serialNumberParam;
```
Comment (KarolinaKopacz) [inline test/features/support/voucher_cancelation.js:6]: I would change `serialNumberParam` to the `voucherSerialNumber`

**Review by KarolinaKopacz — CHANGES_REQUESTED** [review]

File: test/features/support/voucher_cancelation.js [hunk 5]
```diff
@@ -0,0 +1,129 @@
+const pactum = require('pactum');
+const { Given, When, Then, Before, After } = require('@cucumber/cucumber');
+const { localhost } = require('./helpers/helpers');
+
+let specVoucherCancelation;
+let voucherSerialNumber;
+let voucherCancelationBodyBody;
```
Comment (KarolinaKopacz) [inline test/features/support/voucher_cancelation.js:7]:
```suggestion
let voucherCancelationBody;
```

File: test/features/support/voucher_cancelation.js [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+  specVoucherCancelation = pactum.spec();
+});
+
+// Scenario: The user successfully cancels a voucher
+Given('The user wants to cancel a voucher', () => {
+  voucherSerialNumber = '1234';
+  voucherCancelationBodyBody = {
+    voucherserialnumber: voucherSerialNumber,
+    Gov_Stack_BB: 'Gov_Stack_BB'
+  };
```
Comment (KarolinaKopacz) [inline test/features/support/voucher_cancelation.js:24]: You are writing the same code three times. This part can be in constants, for example `validVoucherCancelationBody` and then 
`voucherCancelationBodyBody = validVoucherCancelationBody`

**Review by KarolinaKopacz — CHANGES_REQUESTED** [review]

File: test/features/support/voucher_cancelation.js (hunk 6 above)
Comment (mgrochalskisoldevelo) [reply test/features/support/voucher_cancelation.js:24]: good idea, will change it quick

**Review by KarolinaKopacz — APPROVED** [review]

File: test/features/support/voucher_cancelation.js (hunk 2 above)
Comment (KarolinaKopacz) [reply test/features/support/voucher_cancelation.js]: @mgrochalskisoldevelo @dborowiecki can we close this thread? Or still need to add some changes?

Comment (mgrochalskisoldevelo) [reply test/features/support/voucher_cancelation.js]: @KarolinaKopacz  - In my opinion we can close it. Each test case uses now requestFunction with body variable defined in ```@Given``` statement

**Review by dszafranek — APPROVED** [review]

**Review by oscarcorreia43 — APPROVED** [review]

**Pull request merged**